app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt

```diff
@@ -36,13 +37,8 @@ class DeviceInformationService(
             Configuration.UI_MODE_TYPE_WATCH -> return DeviceType.WATCH
         }
 
-        val windowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
-        val displayMetrics = DisplayMetrics()
-
-        windowManager.defaultDisplay.getRealMetrics(displayMetrics)
-
-        val widthInches = displayMetrics.widthPixels / displayMetrics.xdpi
-        val heightInches = displayMetrics.heightPixels / displayMetrics.xdpi
+        val widthInches = (context.getActivity()?.windowWidth ?: 0).toFloat()
```

Comment:
I think in this case we actually want the `displayMetrics`, as we're not trying to figure out stuff about the window, but instead about the executing device 🙂